src/Tasks.UnitTests/RARPrecomputedCache_Tests.cs

```diff
@@ -0,0 +1,104 @@
+                t.AssemblyInformationCacheOutputPath = precomputedPath;
+                t.WriteStateFile(calculateMvid);
+                File.Exists(standardCache.Path).ShouldBeFalse();
+                int preLen = File.ReadAllText(precomputedPath).Length;
+                preLen.ShouldBeGreaterThan(0);
+                preLen.ShouldNotBe(standardLen);
+            }
+        }
+
+        [Fact]
+        public void TestPreComputedCacheInputAndOutput()
+        {
+            using (TestEnvironment env = TestEnvironment.Create()) {
+                TransientTestFile standardCache = env.CreateFile(".cache");
+                ResolveAssemblyReference t = new ResolveAssemblyReference();
+                t._cache = new SystemState();
+                t._cache.instanceLocalFileStateCache = new Dictionary<string, SystemState.FileState>() {
+                    { Path.Combine(standardCache.Path, "assembly1"), new SystemState.FileState(DateTime.Now) },
+                    { Path.Combine(standardCache.Path, "assembly2"), new SystemState.FileState(DateTime.Now) { Assembly = new Shared.AssemblyNameExtension("hi") } } };
+                t.StateFile = standardCache.Path;
+                t._cache.IsDirty = true;
+                t.WriteStateFile(calculateMvid);
+
+                string dllName = Path.Combine(Path.GetDirectoryName(standardCache.Path), "randomFolder", "dll.dll");
+                t._cache.instanceLocalFileStateCache.Add(dllName,
+                    new SystemState.FileState(DateTime.Now) {
+                        Assembly = null,
+                        RuntimeVersion = "v4.0.30319",
+                        FrameworkNameAttribute = new System.Runtime.Versioning.FrameworkName(".NETFramework", Version.Parse("4.7.2"), "Profile"),
+                        scatterFiles = new string[] { "first", "second" } });
+                string precomputedCachePath = standardCache.Path + ".cache";
+                t.AssemblyInformationCacheOutputPath = precomputedCachePath;
+                t._cache.IsDirty = true;
+                t.WriteStateFile(calculateMvid);
+
+                ResolveAssemblyReference u = new ResolveAssemblyReference();
+                u.StateFile = standardCache.Path;
+                u.AssemblyInformationCachePaths = new ITaskItem[]
+                {
+                    new TaskItem(precomputedCachePath)
+                };
+
+                u.ReadStateFile(File.GetLastWriteTime, Array.Empty<AssemblyTableInfo>(), calculateMvid, p => true);
+                u._cache.instanceLocalFileStateCache.ShouldNotContainKey(dllName);
+                File.Delete(standardCache.Path);
+                u._cache = null;
+                u.ReadStateFile(File.GetLastWriteTime, Array.Empty<AssemblyTableInfo>(), calculateMvid, p => true);
+                u._cache.instanceLocalFileStateCache.ShouldContainKey(dllName);
+                SystemState.FileState a3 = u._cache.instanceLocalFileStateCache[dllName];
```

Comment:
nit: `t`, `u`, and `a3` aren't great variable names.

Reply:
+1

---
In reply to: [505966901](https://github.com/dotnet/msbuild/pull/5514#discussion_r505966901) [](ancestors = 505966901)